### PR TITLE
Mtomaizy

### PR DESCRIPTION
**Pull request by mtomaizy**
add new classes multipartDownloader

### CODE REVIEW

**Comment by GhalebKhaled** [comment]
I think you should consider better names for pull requests

File: src/com/bzu/softwareconstruction/segments/SegmentPartUrl.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public String getItemName() {
 	
 	private void insureStreamOpened(){
 		if(contentType == null){
-			throw new IllegalStateException("You must openStream() before requesting contentType!!");
+			//throw new IllegalStateException("You must openStream() before requesting contentType!!");
```
Comment (GhalebKhaled) [inline src/com/bzu/softwareconstruction/segments/SegmentPartUrl.java:64]: why you commenting this line?

File: src/com/bzu/softwareconstruction/services/Services.java [hunk 2]
```diff
@@ -1,8 +0,0 @@
-package com.bzu.softwareconstruction.services;
```
Comment (GhalebKhaled) [inline src/com/bzu/softwareconstruction/services/Services.java:1]: why you deleted this file?

File: src/com/bzu/softwareconstruction/services/multipart/MultiPartDownloaderImpl.java [hunk 3]
```diff
@@ -1,13 +0,0 @@
-package com.bzu.softwareconstruction.services.multipart;
```
Comment (GhalebKhaled) [inline src/com/bzu/softwareconstruction/services/multipart/MultiPartDownloaderImpl.java:1]: why you deleted this file?

File: src/com/bzu/softwareconstruction/services/Services.java (hunk 2 above)
Comment (mtomaizy) [inline src/com/bzu/softwareconstruction/services/Services.java:1]: by wrong

**Pull request merged**